ott/core/sinkhorn_lr.py

```diff
@@ -426,34 +422,27 @@ def lr_costs(
       self, ot_prob: linear_problems.LinearProblem, state: LRSinkhornState,
       iteration: int
   ) -> Tuple[jnp.ndarray, jnp.ndarray, jnp.ndarray]:
+    log_q, log_r, log_g = jnp.log(state.q), jnp.log(state.r), jnp.log(state.g)
     grad_q = ot_prob.geom.apply_cost(state.r, axis=1) / state.g[None, :]
-    grad_q = jnp.where(
-        self.epsilon != 0., grad_q + self.epsilon * jnp.log(state.q), grad_q
-    )
-    if self.gamma_init == "rescale":
+    grad_q = jnp.where(self.is_entropic, grad_q + self.epsilon * log_q, grad_q)
+    if self.gamma_rescale:
       norm_q = jnp.max(jnp.abs(grad_q)) ** 2
     grad_r = ot_prob.geom.apply_cost(state.q) / state.g[None, :]
-    grad_r = jnp.where(
-        self.epsilon != 0., grad_r + self.epsilon * jnp.log(state.r), grad_r
-    )
-    if self.gamma_init == "rescale":
+    grad_r = jnp.where(self.is_entropic, grad_r + self.epsilon * log_r, grad_r)
+    if self.gamma_rescale:
       norm_r = jnp.max(jnp.abs(grad_r)) ** 2
     diag_qcr = jnp.sum(
         state.q * ot_prob.geom.apply_cost(state.r, axis=1), axis=0
     )
-    grad_g = -diag_qcr / state.g ** 2
-    grad_g = jnp.where(
-        self.epsilon != 0., grad_g + self.epsilon * jnp.log(state.g), grad_g
-    )
-    if self.gamma_init == "rescale":
+    grad_g = -(diag_qcr) / (state.g ** 2)
+    grad_g = jnp.where(self.is_entropic, grad_g + self.epsilon * log_g, grad_g)
+    if self.gamma_rescale:
       norm_g = jnp.max(jnp.abs(grad_g)) ** 2
-    if self.gamma_init == "rescale":
-      self.gamma = self.gamma / max(norm_q, norm_r, norm_g)
-    h = diag_qcr / state.g ** 2 - (self.epsilon -
-                                   1 / self.gamma) * jnp.log(state.g)
-    c_q = grad_q - (1 / self.gamma) * jnp.log(state.q)
-    c_r = grad_r - (1 / self.gamma) * jnp.log(state.r)
-    h = -grad_g + (1 / self.gamma) * jnp.log(state.g)
+    if self.gamma_rescale:
```

Comment:
I'd sligthly prefer to compute all the gradient norms in this `if` statement, it's more readable.